plugin/src/main/java/com/denizenscript/denizen/events/player/PlayerChangesMainHandScriptEvent.java

```diff
@@ -0,0 +1,84 @@
+package com.denizenscript.denizen.events.player;
+
+import com.denizenscript.denizen.objects.*;
+import com.denizenscript.denizen.utilities.implementation.BukkitScriptEntryData;
+import com.denizenscript.denizen.events.BukkitScriptEvent;
+import com.denizenscript.denizencore.objects.core.ElementTag;
+import com.denizenscript.denizencore.objects.core.ListTag;
+import com.denizenscript.denizencore.objects.ObjectTag;
+import com.denizenscript.denizencore.scripts.ScriptEntryData;
+import com.denizenscript.denizencore.utilities.CoreUtilities;
+import org.bukkit.block.BlockState;
+import org.bukkit.block.Sign;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.block.SignChangeEvent;
+import org.bukkit.event.player.PlayerChangedMainHandEvent;
+import org.bukkit.inventory.MainHand;
+
+import java.util.Arrays;
+
+public class PlayerChangesMainHandScriptEvent extends BukkitScriptEvent implements Listener {
+
+    // <--[event]
+    // @Events
+    // player changes main hand
+    //
+    // @Regex ^on player changes main hand$
+    //
+    // @Group Player
+    //
+    // @Triggers when a player changes their main hand.
+    //
+    // @Context
+    // <context.old_hand> returns the player's old main hand, either LEFT or RIGHT.
+    // <context.new_hand> returns the player's new main hand.
+    //
+    // @Player Always.
+    //
+    // -->
+
+    public PlayerChangesMainHandScriptEvent() {
+        instance = this;
+    }
+
+    public static PlayerChangesMainHandScriptEvent instance;
+    public PlayerChangedMainHandEvent event;
+
+    @Override
+    public boolean couldMatch(ScriptPath path) {
+        if (!path.eventLower.startsWith("player changes main hand")) {
```

Comment:
You used an outdated reference event - this couldMatch shouldn't be here, nor should the `regex` in the meta